### PR TITLE
Move NATS module from ballerina-lang

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
$title
Merge https://github.com/ballerina-platform/module-ballerinax-nats/pull/3 before this PR.
Fixes: https://github.com/ballerina-platform/ballerina-lang/issues/24801

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by BuddhiWathsala** [comment]
It seems still there are some language server test failures. We can proceed with this once we completed the test implementations here https://github.com/ballerina-platform/module-ballerinax-nats.

File: language-server/modules/langserver-core/src/test/resources/extensions/document/ast/modify/serviceNats.bal [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+//        methods: ["GET"],
+//        path: "/sayHello"
+//    }
+//    resource function sayHello(http:Caller caller, http:Request req) {
+//        nats:Connection connection = new ();
+//        nats:Producer producer = new (connection);
+//        nats:Error? result = producer->publish("Foo", "Test Message");
+//
+//    }
+//}
```
Comment (BuddhiWathsala) [inline language-server/modules/langserver-core/src/test/resources/extensions/document/ast/modify/serviceNats.bal:17]: Any reason to comment out these NATS related things rather removing?

Comment (aashikam) [reply language-server/modules/langserver-core/src/test/resources/extensions/document/ast/modify/serviceNats.bal:17]: https://github.com/ballerina-platform/ballerina-lang/issues/25183

Comment (aashikam) [reply language-server/modules/langserver-core/src/test/resources/extensions/document/ast/modify/serviceNats.bal:17]: I've created an issue to track it. @nadeeshaan will look into it.

Comment (BuddhiWathsala) [reply language-server/modules/langserver-core/src/test/resources/extensions/document/ast/modify/serviceNats.bal:17]: Ok, thanks.

**Pull request merged**